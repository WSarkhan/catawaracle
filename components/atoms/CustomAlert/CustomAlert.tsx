import { Modal, Text, TouchableOpacity, View } from 'react-native';

interface Props {
  visible: boolean;
  title: string;
  message: string;
  onClose: () => void;
  buttonText?: string;
}

export const CustomAlert = ({
  visible,
  title,
  message,
  onClose,
  buttonText = 'OK',
}: Props) => {
  if (!visible) return null;
  return (
    <Modal transparent animationType="fade" visible={visible}>
      <View className="items-center justify-center flex-1 bg-black/50">
        <View className="w-4/5 p-5 bg-white rounded-lg shadow-md">
          <Text className="text-lg font-bold text-gray-800">{title}</Text>
          <Text className="mt-2 text-sm text-gray-600">{message}</Text>
          <TouchableOpacity
            onPress={onClose}
            className="p-3 mt-5 bg-orange-500 rounded-lg"
          >
            <Text className="font-medium text-center text-white">
              {buttonText}
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};
