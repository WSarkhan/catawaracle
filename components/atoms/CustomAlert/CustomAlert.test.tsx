import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { CustomAlert } from './CustomAlert';

describe('CustomAlert', () => {
  it('should match a snapshot', () => {
    const tree = renderer
      .create(
        <CustomAlert
          title="Title"
          message="Message"
          visible={true}
          onClose={() => {}}
          buttonText="OK"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
