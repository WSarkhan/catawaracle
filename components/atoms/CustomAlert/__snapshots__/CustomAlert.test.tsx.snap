// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert should match a snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    className="items-center justify-center flex-1 bg-black/50"
  >
    <View
      className="w-4/5 p-5 bg-white rounded-lg shadow-md"
    >
      <Text
        className="text-lg font-bold text-gray-800"
      >
        Title
      </Text>
      <Text
        className="mt-2 text-sm text-gray-600"
      >
        Message
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          className="font-medium text-center text-white"
        >
          OK
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
